### ISSUE TITLE
Duplicate synonyms or not or where?

### ISSUE THREAD

**Issue by matuskalas** [edamontology/edamontology#233]
Please first refer to the comments in #170 and https://github.com/edamontology/edamxpathvalidator/issues/1

This issue is about how we should handle duplicate/multiplicate synonyms (allow or not; between sub-ontologies or within; exact or which synonyms).

And it needs expert text mining practitioners to advise.

**Comment by joncison** [comment]
This is a validation issue, @hmenager please move this issue to https://github.com/edamontology/edamxpathvalidator

**Comment by matuskalas** [comment]
This is not a validation issue. For the corresponding validation issue, please see edamontology/edamxpathvalidator#1. This issue is about maintaining the actual synonyms in EDAM.

**Comment by joncison** [comment]
Will this not boil down to a set of rules that should be encoded in the validator?  In which case I think it should be moved.  Once we have the rules then an issue in this tracker could be created to fix the problems identified.

**Comment by joncison** [comment]
@matuskalas I propose (for simplicity, practicality)

* disallow duplicate synonyms within a subontology
* allow duplicate synonyms between subontologies
* treat all types (narrow, broad, exact) the same

no doubt there's a few outliers but we can see these once the QC is in place

what do you think?

**Comment by matuskalas** [comment]
I agree with the 1st and 3rd rules. Skeptical whether the 2nd is a good idea.

**Comment by joncison** [comment]
@hmenager - can you give an update on whether the above checks are implemented in the QC ? 

- disallow duplicate synonyms within a subontology - raise ERROR
- duplicate synonyms between subontologies - raise WARN

We'd also need checks of labels versus synonyms too, if these are not already done:
- label:synonym duplicates within subontologies - raise ERROR
- label:synyonym duplicates between subontologies - raise WARN

And also
- label:label duplicates within subontologies - raise ERROR
- label:label duplicates between subontologies - raise WARN

(putting all types of synonyms into one bag for this purpose)

cc @matuskalas FHI

**Comment by joncison** [comment]
@matuskalas I'm  closing this now, because I think we agree.

As for *allow duplicate synonyms between subontologies* - I'm firmly persuaded that, as a rule we should avoid this, but it is a necessary evil, *i.e.* so we can have *e.g.* the **Operation** *Sequence alignment* and the **Data** *Sequence alignment* (the alternative is bloat/ugly/worse)

What's left to do elsewhere:
* https://github.com/edamontology/edamxpathvalidator/issues/3
* https://github.com/edamontology/edamxpathvalidator/issues/1
* https://github.com/edamontology/edamontology/issues/354
